apis/v1/policySnapshot_types.go

```diff
@@ -0,0 +1,134 @@
+}
+
+// PolicySnapShotSpec defines the desired state of PolicySnapShot.
+type PolicySnapShotSpec struct {
+	// Policy defines how to select member clusters to place the selected resources.
+	// If unspecified, all the joined member clusters are selected.
+	// +optional
+	Policy *PlacementPolicy `json:"policy,omitempty"`
+
+	// IsLatest indicates if the policy is the latest or not.
+	// +required
+	IsLatest bool `json:"isLatest"`
+
+	// PolicyHash is the sha-256 hash value of the Policy field
+	// +required
+	PolicyHash []byte `json:"policyHash"`
+}
+
+// PolicySnapShotStatus defines the observed state of PolicySnapShot.
+type PolicySnapShotStatus struct {
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+
+	// Conditions is an array of current observed conditions for PolicySnapShot.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// ClusterDecisions contains a list of names of member clusters considered by the scheduler.
+	// Note that all the selected clusters must present in the list while not all the
+	// member clusters are guaranteed to be listed.
+	// +optional
+	ClusterDecisions []ClusterDecision `json:"targetClusters,omitempty"`
+}
+
+// PolicySnapShotConditionType identifies a specific condition of the PolicySnapShot.
+type PolicySnapShotConditionType string
+
+const (
+	// 	Scheduled indicates the scheduled condition of the given policySnapShot.
+	// Its condition status can be one of the following:
+	// - "True" means the corresponding policySnapShot is fully scheduled.
+	// - "False" means the corresponding policySnapShot is not scheduled yet.
+	// - "Unknown" means this policy does not have a full schedule yet.
+	Scheduled PolicySnapShotConditionType = "Scheduled"
+)
+
+// ClusterDecision represents a decision from a placement
+// An empty ClusterDecision indicates it is not scheduled yet.
+type ClusterDecision struct {
+	// ClusterName is the name of the ManagedCluster. If it is not empty, its value should be unique cross all
+	// placement decisions for the Placement.
+	// +kubebuilder:validation:Required
+	// +required
+	ClusterName string `json:"clusterName"`
+
+	// Selected indicates if this cluster is selected by the scheduler.
+	// +required
+	Selected bool `json:"selected"`
+
+	// AffinityScore represents the affinity score of the cluster calculated by the last
+	// scheduling decision based on the preferred affinity selector.
+	// An affinity score may not present if the cluster does not meet the required affinity.
+	// +optional
+	AffinityScore int32 `json:"affinityScore,omitempty"`
+
+	// TopologySpreadScore represents the priority score of the cluster calculated by the last
+	// scheduling decision based on the topology spread applied to the cluster.
+	// A priority score may not present if the cluster does not meet the topology spread.
+	// +optional
+	TopologySpreadScore int32 `json:"priorityScore,omitempty"`
```

Comment:
should we create a separate structure, eg ClusterScore for them?

later, it will be easy to extend.

Reply:
good point